nanopub/client.py

```diff
@@ -201,13 +202,13 @@ def publish(self, nanopub: Nanopub):
 
         if nanopub.introduces_concept:
             concept_uri = str(nanopub.introduces_concept)
-            # Replace the DEFAULT_URI with the actually published nanopub uri. # This is
+            # Replace the DEFAULT_NANOPUB_URI with the actually published nanopub uri. # This is
             # necessary if a blank node was passed as introduces_concept. In that case the
             # Nanopub.from_assertion method replaces the blank node with the base nanopub's URI
             # and appends a fragment, given by the 'name' of the blank node. For example, if a
             # blank node with name 'step' was passed as introduces_concept, the concept will be
             # published with a URI that looks like [published nanopub URI]#step.
-            concept_uri = concept_uri.replace(DEFAULT_URI, nanopub_uri)
+            concept_uri = concept_uri.replace(DEFAULT_NANOPUB_URI, nanopub_uri)
```

Comment:
I feel like at some point we should move to using named blank nodes for this stuff as well. In ```from_assertion```, for example, an rdflib blank node is automatically converted to ```NANOPUB_URI#BLANK_NODE_NAME```. Ideally we'd find a way to stop the user setting the nanopub URI at all, since it just leads to confusion because it's a dummy URI... It's more for historical reasons that the user can still set that. Any ideas?

Reply:
I've created #38